src/pepr/operator/controllers/network/generate.ts

```diff
@@ -0,0 +1,145 @@
+import { V1APIGroup, V1NetworkPolicyPeer, V1NetworkPolicyPort } from "@kubernetes/client-node";
+import { K8s, Log, kind } from "pepr";
+
+import { Allow, UDSPackage } from "../../crd";
+import { RemoteGenerated } from "../../crd/generated/package-v1alpha1";
+
+let apiServerPeers: V1NetworkPolicyPeer[];
+
+export async function generate(
+  namespace: string,
+  pkg: UDSPackage,
+  policy: Allow,
+  idx: number,
+): Promise<kind.NetworkPolicy> {
+  const pkgName = pkg.metadata!.name!;
+  const target = Object.values(policy.podLabels || ["all-pods"]).join("-");
+
+  // Create a unique name for the NetworkPolicy based on the package name, index, direction, pod labels, and port
+  const name = `allow-${policy.direction}-${target}-${pkgName}-${idx}`.toLowerCase();
```

Comment:
I also added "uds" to this name after this screenshot, but thik this might make more sense? 

<img width="1486" alt="Screenshot 2024-01-04 at 1 40 04 PM" src="https://github.com/defenseunicorns/uds-core/assets/882485/e0c27ebc-d817-48bc-b43e-12c80641bf86">

Reply:
Yeah I like that much better, makes the numbers make a bit more sense and will order them nicely in outputs.